libraries/botbuilder-azure/botbuilder/azure/cosmosdb_storage.py

```diff
@@ -160,20 +160,30 @@ def __init__(
         :param changes:
         :return:
         """
+        if changes is None:
+            raise Exception("Changes are required when writing")
+        if not changes:
+            return
         try:
             # check if the database and container exists and if not create
             if not self.__container_exists:
                 self.__create_db_and_container()
                 # iterate over the changes
             for (key, change) in changes.items():
                 # store the e_tag
-                e_tag = change.e_tag
+                e_tag = (
+                    change.e_tag
+                    if hasattr(change, "e_tag")
+                    else change.get("e_tag", None)
+                )
                 # create the new document
                 doc = {
                     "id": CosmosDbKeyEscape.sanitize_key(key),
                     "realId": key,
                     "document": self.__create_dict(change),
                 }
+                if e_tag == "":
```

Comment:
if not e_tag

Reply:
Same as previous. We want to allow e_tag to be None so it can act as a wildcard, but we don't want the e_tag to be blank.

Reply:
Same as previous. We want to allow e_tag to be None so it can act as a wildcard, but we don't want the e_tag to be blank.